### ISSUE TITLE
Text-Shadow bug

### ISSUE THREAD

**Issue by Janne-d-Arc** [linuxmint/Cinnamon#4483]
When the wallpaper is too bright, Cinnamon does not show shadows under the desklet's text unless I hover the mouse pointer over it, making the text difficult to read.

This is how the text looks when the system starts:

![before](https://cloud.githubusercontent.com/assets/13596575/9020586/214e792c-37db-11e5-8422-0c02e61e1d89.png)

And this is how the text looks when I hover the mouse pointer over it:

![after](https://cloud.githubusercontent.com/assets/13596575/9020587/2ff93796-37db-11e5-85b1-94b5d0096029.png)

**Comment by Vahan86** [comment]
@Janne-d-Arc, is this still an issue for you in Cinnamon 3.0?

**Comment by Vahan86** [comment]
@JosephMcc, can we close the issue?

**Comment by germanfr** [comment]
I can't reproduce it on 3.2. But there's another issue with the shadow. It is not drawn in half of the last digit:
![screenshot from 2016-11-19 20-35-16](https://cloud.githubusercontent.com/assets/10391266/20457960/ef52b5d4-ae97-11e6-8116-254f991cadb8.png)
![screenshot from 2016-11-19 20-34-47](https://cloud.githubusercontent.com/assets/10391266/20457961/ef541c4e-ae97-11e6-918a-7adda2517feb.png)

**Comment by JosephMcc** [comment]
This issue needs some better info supplied. What desklet are you using? What are the settings? What Cinnamon version? What theme?

**Comment by germanfr** [comment]
Yes, I think it's a theme issue. It's weird, I've never seen a cut shadow in css. I've tried it with the default theme and Mint-Y (adding a shadow) and there is no problem there. I'm sorry.

If you still interested it is the clock desklet, Cinnamon 3.2.1 and "New-Minty" theme.

**Comment by JosephMcc** [comment]
Thanks for following up. Best bet would probably be to see if you can find the difference between the themes.

**Comment by germanfr** [comment]
I've reproduced the shadow cut on the default theme. It seems it's not a theme issue.

It happens when you apply a css padding to a StLabel. In this case, it happended when applying it to `clock-desklet-label`.
![screenshot from 2016-11-25 20-05-58](https://cloud.githubusercontent.com/assets/10391266/20633823/e1b1b742-b34a-11e6-8798-18af3d9897d0.png)

Should I open another issue? It is not a top priority problem anyway :P

**Comment by JosephMcc** [comment]
Nope, I'll reopen this one.

**Comment by germanfr** [comment]
I've discovered it affects elements other than the clock, like for example this in the menu applet:
![screenshot from 2017-02-01 17-26-38](https://cloud.githubusercontent.com/assets/10391266/22515837/32e1c04c-e8a4-11e6-8469-8d5115139233.png)

**Comment by JosephMcc** [comment]
@germanfr Just made a PR to fix the problem you are seeing. The original problem however persists. In many cases text shadows are not showing up on cinnamon elements until the first time they are hovered. After that they are fine. I can see the problem in both panel applets and desklets.

**Comment by Janne-d-Arc** [comment]
The problem still persist in LM 18.2 and Cinnamon 3.4.6, in LM 17 with Cinnamon 2.2 it was working fine.

**Comment by Janne-d-Arc** [comment]
The problem still persist in LM 18.3 and Cinnamon 3.6.7, it seems that the problem is because Cinnamon doesn't apply the CSS file until the mouse pointer is hovered over the desklet.

**Comment by Mumrik93** [comment]
I personally have a problem with both desklets and the general text on my desktop not having text-shadows (which really strains my eyes) and i cant find a way to fix it. Really wish the Linux Mint team would increase it a few px's!

**Comment by Vahan86** [comment]
@Janne-d-Arc, is this still an issue in Cinnamon 4.2?

**Comment by Mumrik93** [comment]
> @Janne-d-Arc, is this still an issue in Cinnamon 4.2?

I cant comment on the desklet issues, but for the text under the desktop-icons, it can be solved by following any of the options by the end of this thread;
https://forums.linuxmint.com/viewtopic.php?f=90&t=282853&start=20

Sadly those shadows aren't included by default for some reason.. dont understand why.

**Comment by Janne-d-Arc** [comment]
Sorry for the late reply, I confirm that in the current version of Cinnamon (4.4.8) this still happens but very rarely, I even thought it was already fixed but it still happens from time to time. Previously, the CSS file was never applied until the mouse pointer was hovered over the Desklet, now it works fine almost every time and there is no need to do that.
